packages/shared/src/devBrowser.ts

```diff
@@ -19,20 +19,48 @@ export function setDevBrowserJWTInURL(url: URL, jwt: string): URL {
   return resultURL;
 }
 
-// Gets the dev_browser JWT from either the hash or the search
-// Side effect:
-// Removes dev_browser JWT from the URL as a side effect and updates the browser history
-export function getDevBrowserJWTFromURL(url: URL): string {
-  const resultURL = new URL(url);
-
-  // extract & strip existing jwt from search
-  const jwt = resultURL.searchParams.get(DEV_BROWSER_JWT_KEY) || '';
-  resultURL.searchParams.delete(DEV_BROWSER_JWT_KEY);
-
-  // eslint-disable-next-line valid-typeof
-  if (jwt && typeof globalThis.history !== undefined) {
-    globalThis.history.replaceState(null, '', resultURL.href);
+/**
+ * Gets the __clerk_db_jwt JWT from either the hash or the search
+ * Side effect:
+ * Removes __clerk_db_jwt JWT from the URL (hash and searchParams) and updates the browser history
+ */
+export function extractDevBrowserJWTFromURL(url: URL): string {
+  const jwt = readDevBrowserJwtFromSearchParams(url);
+  if (jwt && typeof globalThis.history !== 'undefined') {
+    globalThis.history.replaceState(null, '', removeDevBrowserJwt(url));
   }
-
   return jwt;
 }
+
+const readDevBrowserJwtFromSearchParams = (url: URL) => {
+  return url.searchParams.get(DEV_BROWSER_JWT_KEY) || '';
+};
+
+const removeDevBrowserJwt = (url: URL) => {
+  return removeDevBrowserJwtFromURLSearchParams(removeLegacyDevBrowserJwtFromURLHash(new URL(url)));
+};
+
+const removeDevBrowserJwtFromURLSearchParams = (_url: URL) => {
+  const url = new URL(_url);
+  url.searchParams.delete(DEV_BROWSER_JWT_KEY);
+  return url;
+};
```

Comment:
☁️ I think we should follow the same pattern in our methods. I think we should adhere to the pattern where we construct a new instance and return the updated cloned instance instead of mutating the parameters passed.
This is a preferred pattern as it helps with improving visibility of the changes instead of hiding them which could cause unwanted side-effects and hard-to-find bugs.

`removeDevBrowserJwtFromURLSearchParams` returns a new url instance with the updated query params vs
`removeLegacyDevBrowserJwtFromURLHash` mutates the existing url instance.

Reply:
Good point, I updated the code 👍🏻